src/bstr.rs

```diff
@@ -60,3 +60,67 @@ impl BStr {
         &self.bytes
     }
 }
+
+/// Construct a literal `BStr` usable in `static` or `const` items.
+///
+/// # Examples
+///
+/// ## Basic usage
+///
+/// Note that a byte string literal must be used, e.g. `b"foobar"` and not
+/// `"foobar"`.
+///
+/// ```
+/// # use bstr::{B, BStr};
+/// const EXAMPLE: &BStr = bstr::literal!(b"foobar");
+/// assert_eq!(EXAMPLE, B("foobar"));
+/// ```
+///
+/// ## Use in combination with `include_bytes!`
+///
+/// ```
+/// # const _: &str = stringify! {
+/// const FILE_BYTES: &BStr = bstr::literal!(include_bytes!("./file.dat"));
+/// # };
+/// ```
+///
+/// ## Populating a field in a static struct
+///
+/// ```rust
+/// # #[cfg(feature = "std")]
+/// # fn main() {
+/// # use std::borrow::Cow;
+/// # use bstr::{B, BStr};
+///
+/// struct Archive {
+///     name: Cow<'static, str>,
+///     raw: Cow<'static, BStr>,
```

Comment:
I don't think I quite grok the significance of using `Cow` here? Why not use `&'static str` and `&'static BStr` instead?